client/src/components/TypeList.vue

```diff
@@ -126,6 +127,14 @@ export default defineComponent({
       setCheckedTypes([]);
     }
 
+    const typeCounts = computed(() => filteredTracksRef.value.reduce((acc, filteredTrack) => {
+      const type = filteredTrack.track.getType(filteredTrack.context.confidencePairIndex);
+      const trackType = type ? type[0] : '';
+      acc.set(trackType, (acc.get(trackType) || 0) + 1);
+
+      return acc;
+    }, new Map<string, number>()));
```

Comment:
Reduce is definitely the best vanilla js option.  Another option is `lodash.groupBy` since we already have it for an even simpler version:

https://lodash.com/docs/4.17.15#groupBy

``` js
import { groupBy } from 'lodash';
// ...
const typeCounts = computed(() => groupBy(
    filteredTracksRef.value,
    (v) => v.track.getType(v.context.confidencePairIndex)
);
```

The only difference is now you have to use `.length` on the map values in the template.

Also, I'm not sure what the purpose of line 132 is.  `track.getType` does not return a nullable value.  `type[0]` and type[1]` always exist or the function throws an exception.

Reply:
Removed the extraneous null check.

As we discussed earlier, it is best to keep the `reduce` implementation for type counts to keep the code easy to understand.